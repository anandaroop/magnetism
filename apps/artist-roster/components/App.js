import React, { Component } from 'react'

class App extends Component {
  render() {
    return <div>I'm a properly linted App!</div>
  }
}

export default App
