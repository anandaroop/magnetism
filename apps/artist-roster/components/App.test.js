import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import App from './App'

describe('App', () => {
  it('renders', () => {
    const component = renderer.create(<App />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('responds to clicks', () => {
    const component = shallow(<App />)
    expect(component.state('n')).toEqual(1)
    expect(toJson(component)).toMatchSnapshot()
    
    component.find('a').simulate('click', { preventDefault: jest.fn() })
    expect(component.state('n')).toEqual(2)
    expect(toJson(component)).toMatchSnapshot()
  })
})
