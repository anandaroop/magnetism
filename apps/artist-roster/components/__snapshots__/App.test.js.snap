// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<div>
  <h1
    className="sc-bdVaJa dJdHpx"
  >
    Hi Styles
  </h1>
  <p>
    Do you like numbers? Great!
     
    <a
      href="#"
      onClick={[Function]}
    >
      Add more
    </a>
  </p>
  <ul
    className="sc-bwzfXH fGIOEA"
  >
    <li
      className="sc-htpNat cgsvxu"
    >
      0
    </li>
  </ul>
</div>
`;

exports[`App responds to clicks 1`] = `
<div>
  <styled.h1>
    Hi Styles
  </styled.h1>
  <p>
    Do you like numbers? Great!
     
    <a
      href="#"
      onClick={[Function]}
    >
      Add more
    </a>
  </p>
  <styled.ul>
    <styled.li>
      0
    </styled.li>
  </styled.ul>
</div>
`;

exports[`App responds to clicks 2`] = `
<div>
  <styled.h1>
    Hi Styles
  </styled.h1>
  <p>
    Do you like numbers? Great!
     
    <a
      href="#"
      onClick={[Function]}
    >
      Add more
    </a>
  </p>
  <styled.ul>
    <styled.li>
      0
    </styled.li>
    <styled.li>
      1
    </styled.li>
  </styled.ul>
</div>
`;
